litestar/openapi/spec/media_type.py

```diff
@@ -32,7 +32,7 @@ class OpenAPIMediaType(BaseSchemaObject):
     example provided by the schema.
     """
 
-    examples: dict[str, Example | Reference] | None = None
+    examples: dict[str, Example | Reference] | Mapping[str, Example] | None = None
```

Comment:
While I think this is a good change, it's unfortunately breaking; If someone is using this attribute and referring to dict-only methods on it, passing a different type of mapping here would break their code and also their type checking. 

Iiuc having this be a `Mapping` isn't really necessary for the proposed feature, right?